### DISCUSSION TITLE
why doesn't `rg -F "$finish"` (double quotes) show the same results as `rg -F '$finish'` (single quotes)?

### DISCUSSION THREAD

**Discussion by abhijithcd** [BurntSushi/ripgrep#2697] [Q&A] [answered]
### Please tick this box to confirm you have reviewed the above.

- [X] I have a different issue.

### What version of ripgrep are you using?

Using v13.0.0.

### How did you install ripgrep?

linux-musl

### What operating system are you using ripgrep on?

CentOS 7.6,  zsh 5.0.2

### Describe your bug.

Varying behaviour when searching a string with '$' character in it.

Searching `rg -F '$finish'` and `rg '\$finish'` works fine.
Issue cases reported below.

I saw a few similar issues reported for windows but didn't see it for linux. 
Is this an expected behaviour?

### What are the steps to reproduce the behavior?

`rg "\$finish"` and  `rg -F "$finish"`

### What is the actual behavior?

Searching with`rg "\$finish"` doesn't return any results

Searching with `rg -F "$finish"` dumps a huge amount of text (content of all the files under the search path).

### What is the expected behavior?

ripgrep should return 2 matches when using double quotes as well.

**Comment by s-p-turner** [comment, marked as answer]
This looks like expected behaviour to me. I don't believe your issue is anything to do with ripgrep. The dollar sign is a special character which Linux uses to specify an environment variable. Putting the dollar sign in double quotes (or no quotes) causes the Linux shell interpreter to substitute the name of the environment variable with its value. Specifying _\\$_ stops the shell from doing the substitution, and the literal _$_ character is passed instead. Putting the dollar sign in single quotes stops the shell interpreter from doing this environment variable detection.

So -

1. '$finish'  - passes the literal string **$finish**.
2. '\\$finish' - passes the literal string **\\$finish**.
3. "$finish"  - passes the value of the environment variable named _finish_, which will be blank if the environment variable is unassigned.
4. "\\$finish" - passes the literal string **$finish** (same as no. 1)

**Reply by BurntSushi** [reply]
Great answer! Thank you. Two small clarifications:

1. This behavior isn't really a property of Linux, but rather, of the shell. I know you kind of mentioned this, but I wanted to make it clear. You can get the same behavior on Windows by using bash there, for example.
2. Note that `rg "\$finish"` and `rg '\$finish'` are two different things. The first searches for the regex pattern `$finish` (which can never match anything) while the latter searches for the regex pattern `\$finish`. Basically, using single quotes lets you opt out of the shell's parameter expansion logic, and thus you only need to deal with the regex pattern language. But with double quotes, you have to deal with the regex pattern language _inside of the shell's parameter expansion language_.

The TL;DR is that when using a Unix shell, a good rule of thumb is to put a regex pattern in single quotes. My own rule of thumb is that I don't use quotes at all for simple literal searches, but whenever I introduce any "regex" characters into the pattern, then I just wrap it in single quotes.

**Comment by abhijithcd** [comment]
Thank you for the answers. I agree with not putting quotes for the simple searches. I had used them just in case if I wanted to extend my search to mutliple words. 
I'll remember the tip to prefer single quotes.
